contracts/utilityToken/ERC20Token.sol

```diff
@@ -0,0 +1,265 @@
+
+    using SafeMath for uint256;
+
+    /** Name of the token. */
+    string internal tokenName;
+
+    /** Symbol of the token. */
+    string internal tokenSymbol;
+
+    /** Decimals used by the token. */
+    uint8 internal tokenDecimals;
+
+    /** Total supply of the token. */
+    uint256 internal totalTokenSupply;
+
+    /** Stores the token balance of the accounts. */
+    mapping(address => uint256) balances;
+
+    /** Stores the authorization information. */
+    mapping(address => mapping(address => uint256)) allowed;
+
+
+    /* Public Functions. */
+
+    /**
+     * @notice Public function to get the name of the token.
+     *
+     * @return tokenName_ Name of the token.
+     */
+    function name() public view returns (string memory tokenName_) {
+        tokenName_ = tokenName;
+    }
+
+    /**
+     * @notice Public function to get the symbol of the token.
+     *
+     * @return tokenSymbol_ Symbol of the token.
+     */
+    function symbol() public view returns (string memory tokenSymbol_) {
+        tokenSymbol_ = tokenSymbol;
+    }
+
+    /**
+     * @notice Public function to get the decimals of the token.
+     *
+     * @return tokenDecimals Decimals of the token.
+     */
+    function decimals() public view returns (uint8 tokenDecimals_) {
+        tokenDecimals_ = tokenDecimals;
+    }
+
+    /**
+     * @notice Get the balance of an account.
+     *
+     * @param _owner Address of the owner account.
+     *
+     * @return balance_ Account balance of the owner account.
+     */
+    function balanceOf(address _owner) public view returns (uint256 balance_) {
+        balance_ = balances[_owner];
+    }
+
+    /**
+     * @notice Public function to get the total supply of the tokens.
+     *
+     * @dev Get totalTokenSupply as view so that child cannot edit.
+     *
+     * @return totalTokenSupply_ Total token supply.
+     */
+    function totalSupply() public view returns (uint256 totalTokenSupply_) {
+        totalTokenSupply_ = totalTokenSupply;
+    }
+
+    /**
+     * @notice Public function to get the allowance.
+     *
+     * @param _owner Address of the owner account.
+     * @param _spender Address of the spender account.
+     *
+     * @return allowance_ Remaining allowance for the spender to spend from
+     *                    owner's account.
+     */
+    function allowance(address _owner, address _spender)
+        public
+        view
+        returns (uint256 allowance_)
+    {
+        allowance_ = allowed[_owner][_spender];
+    }
+
+    /**
+     * @notice Public function to transfer the token.
+     *
+     * @dev Fires the transfer event, throws if, _from account does not have
+     *      enough tokens to spend.
+     *
+     * @param _to Address to which tokens are transferred.
+     * @param _value Amount of tokens to be transferred.
+     *
+     * @return success_ `true` for a successful transfer, `false` otherwise.
+     */
+    function transfer(address _to, uint256 _value)
```

Comment:
`_amount`

Reply:
I stand corrected, there s good reason to use `value` too; my personal preference is to switch to `amount` but, I'll leave choice to @jayeshbairagi

Reply:
Going with `_value`